sql/connect/common/src/main/protobuf/spark/connect/pipelines.proto

```diff
@@ -90,6 +92,24 @@ message PipelineCommand {
     optional string format = 8;
   }
 
+  // Metadata about why a query function failed to be executed successfully.
+  message QueryFunctionFailure {
+    // Identifier for a dataset within the graph that the query function needed to know the schema
+    // of but which had not yet been analyzed itself.
+    optional string missing_dependency = 1;
```

Comment:
Is is possible for us to return all missing dependencies in one go? That should help us to be more efficient during initialization.